overview.md

```diff
@@ -1,7 +1,7 @@
 # Jupyter Governance Overview
 
 :::{note}
-Jupyter transitioned from a [Benevolent Dictator For Life (BDFL) + Steering Council](governance.md) governance model to this current governance model in December 2022.
+Jupyter transitioned from a [Benevolent Dictator For Life (BDFL) + Steering Council](archive/governance.md) governance model to this current governance model in December 2022.
```

Comment:
A change also needs to be done in the _toc.yml: https://github.com/jupyter/governance/pull/226/files#diff-3cccf3dee374305b8a8af99ba002fba0bf9932305cb9cbdcc81c962cbf54084f

Reply:
Done in https://github.com/jupyter/governance/pull/228/commits/9f60d4845f327af6bd03f0877c9fa51754425e65